ambry-router/src/test/java/com.github.ambry.router/GetBlobInfoOperationTest.java

```diff
@@ -0,0 +1,429 @@
+   * @throws Exception
+   */
+  private void testErrorPrecedence(ServerErrorCode[] serverErrorCodesInOrder, RouterErrorCode expectedErrorCode)
+      throws Exception {
+    GetBlobInfoOperation op =
+        new GetBlobInfoOperation(routerConfig, mockClusterMap, responseHandler, blobIdStr, operationFuture, null, time);
+    ArrayList<RequestInfo> requestListToFill = new ArrayList<>();
+    requestRegistrationCallback.requestListToFill = requestListToFill;
+
+    int i = 0;
+    for (MockServer server : mockServerLayout.getMockServers()) {
+      server.setServerErrorForAllRequests(serverErrorCodesInOrder[i++]);
+    }
+
+    while (!op.isOperationComplete()) {
+      op.poll(requestRegistrationCallback);
+      List<ResponseInfo> responses = sendAndWaitForResponses(requestListToFill);
+      for (ResponseInfo response : responses) {
+        op.handleResponse(response);
+        if (op.isOperationComplete()) {
+          break;
+        }
+      }
+    }
+
+    RouterException routerException = (RouterException) op.getOperationException();
+    Assert.assertEquals(expectedErrorCode, routerException.getErrorCode());
+  }
+
+  /**
+   * Test the case with multiple errors (server level and partition level) from multiple servers,
+   * with just one server returning a successful response. The operation should succeed.
+   * @throws Exception
+   */
+  @Test
+  public void testSuccessInThePresenceOfVariousErrors()
+      throws Exception {
+    GetBlobInfoOperation op =
+        new GetBlobInfoOperation(routerConfig, mockClusterMap, responseHandler, blobIdStr, operationFuture, null, time);
+    ArrayList<RequestInfo> requestListToFill = new ArrayList<>();
+    requestRegistrationCallback.requestListToFill = requestListToFill;
+
+    ArrayList<MockServer> mockServers = new ArrayList<>(mockServerLayout.getMockServers());
+    // set the status to various server level or partition level errors (not Blob_Deleted or Blob_Expired).
+    mockServers.get(0).setServerErrorForAllRequests(ServerErrorCode.Unknown_Error);
+    mockServers.get(1).setServerErrorForAllRequests(ServerErrorCode.Unknown_Error);
+    mockServers.get(2).setServerErrorForAllRequests(ServerErrorCode.IO_Error);
+    mockServers.get(3).setServerErrorForAllRequests(ServerErrorCode.Blob_Not_Found);
+    mockServers.get(4).setServerErrorForAllRequests(ServerErrorCode.Data_Corrupt);
+    mockServers.get(5).setServerErrorForAllRequests(ServerErrorCode.Blob_Not_Found);
+    mockServers.get(6).setServerErrorForAllRequests(ServerErrorCode.Blob_Not_Found);
+    mockServers.get(7).setServerErrorForAllRequests(ServerErrorCode.Disk_Unavailable);
+    mockServers.get(8).setServerErrorForAllRequests(ServerErrorCode.Unknown_Error);
+
+    // set the status of one of the servers in the local datacenter to success (we depend on an actual put of the blob
+    // in order to get back the BlobInfo, and the put would have gone only to the local datacenter).
```

Comment:
Yeah, I think I left this incomplete. Have changed it so that the cross-colo case also gets tested. Basically, we set it this way and then test the case where the get originates from all 3 datacenters, and the blob is found only in DC1.